### PR TITLE
Helm: Multiple topology spread constraints support

### PR DESCRIPTION
**Pull request by jotdl**
#### What this PR does
We have a use case where we need the ability to define multiple topology spread constraints. Kubernetes supports this as documented under https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints/.

Before this PR we only supported on topology spread constraint and this now adds support for multiple ones. Backwards compatibility is kept by supporting the old object/dict syntax next to the new array based syntax as shown in test-oss-topology-spread-constraints-values.yaml.

#### Checklist

- [X] Tests updated
- [ ] Documentation added
- [X] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by jotdl** [comment]
Apologies for the noise (amount of changed files). I tried to improve formatting of the changed yaml sections which affected almost all tests. 

If you think that's too much, I can also try to split this into 1 smaller PRs with the actual enhancement and another PR which would add the formatting improvements.

**Review by dimitarvdimitrov — COMMENTED** [review]
We talked privately about making it possible to specify different label matchers for the different topology spread constraints. This might make sense if someone is trying to spread e.g. store-gateways and ingesters evenly or spread all pods that require persistent storage in their k8s cluster evenly.

If you choose to do this in this PR I think it's important that we don't make the selectors mandatory, so current users can keep using the topology spread constraints as they are today.

File: operations/helm/charts/mimir-distributed/CHANGELOG.md [hunk 1]
```diff
@@ -33,6 +33,7 @@ Entries should include a reference to the Pull Request that introduced the chang
 * [ENHANCEMENT] Update memcached to `memcached:1.6.19-alpine`. #4581
 * [BUGFIX] Include podAnnotations on the tokengen Job. #4540
 * [BUGFIX] Add http port in ingester and store-gateway headless services. #4573
+* [ENHANCEMENT] Allow definition of multiple topology spread constraints. #4584
```
[the remark this answers is not in the document]
Comment (jotdl) [reply operations/helm/charts/mimir-distributed/CHANGELOG.md]: Moved to the correct position!

File: operations/helm/charts/mimir-distributed/ci/offline/test-oss-topology-spread-constraints-values.yaml [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+enterprise:
+  enabled: true
+
+alertmanager:
+  topologySpreadConstraints:
+  - maxSkew: 1
+    topologyKey: kubernetes.io/hostname
+    whenUnsatisfiable: DoNotSchedule
+    matchLabelKeys:
+      - no-alertmanager
```
[the remark this answers is not in the document]
Comment (jotdl) [reply operations/helm/charts/mimir-distributed/ci/offline/test-oss-topology-spread-constraints-values.yaml]: Added support for `labelSelector` and removed `matchLabelKeys` instead.

**Review by dimitarvdimitrov — COMMENTED** [review]
Looks good! Thank you for doing this.

I think the last thing to be done is update the docs in the default values.yaml. Currently they say that the labelSelectors will be generated by the chart, which not always true now. They also don't mention that you can pass multiple topologySpreadConstraints in an array/slice. I think this will be as simple as a single find-and-replace.

**Review by dimitarvdimitrov — APPROVED** [review]
LGTM

**Pull request merged**